### PR TITLE
replace kubernetes fanout request function with class

### PR DESCRIPTION
**Pull request by mclarke47**
Change the fan out mechanism to be a class to fit in more with the style rest of the backend plugin and make DI easier.

### CODE REVIEW

File: plugins/kubernetes-backend/src/service/KubernetesFanOutHandler.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  }
+
+  async getKubernetesObjectsByEntity(
+    requestBody: KubernetesRequestBody,
+    objectTypesToFetch: Set<KubernetesObjectTypes> = DEFAULT_OBJECTS,
+  ) {
+    const entityName = requestBody.entity.metadata.name;
+
+    const clusterDetails: ClusterDetails[] = await this.serviceLocator.getClustersByServiceId(
+      entityName,
```
Comment (freben) [inline plugins/kubernetes-backend/src/service/KubernetesFanOutHandler.ts:60]: Is the locator a per-namespace thing, or are namespaces simply left as a future extension? Big caveat that I may be misunderstanding things :)

Comment (mclarke47) [reply plugins/kubernetes-backend/src/service/KubernetesFanOutHandler.ts:60]: Namespaces are for the future, the serviceLocator is an interface to map a service to which cluster it actually lives on, but ATM the only implementation assumes that every service lives on every cluster

**Review by Rugvip — COMMENTED** [review]
No time for a proper review yet, but please add a changeset :grin: ❤️

**Comment by mclarke47** [comment]
@Rugvip @freben if you get a second 🙏

File: plugins/kubernetes-backend/src/service/KubernetesFanOutHandler.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      return kubernetesAuthTranslator.decorateClusterDetailsWithAuth(
+        cd,
+        requestBody,
+      );
+    });
+    const clusterDetailsDecoratedForAuth: ClusterDetails[] = await Promise.all(
+      promises,
+    );
+
+    this.logger.info(
```
Comment (freben) [inline plugins/kubernetes-backend/src/service/KubernetesFanOutHandler.ts:77]: Drop to debug? Otherwise provide some text about what it does perhaps

**Review by freben — APPROVED** [review]
One suggestion comment, but merging to not block any longer

**Pull request merged**